test/xdr/contract/spec/sc_spec_type_def_test.exs

```diff
@@ -0,0 +1,190 @@
+defmodule StellarBase.XDR.SCSpecTypeDefTest do
+  use ExUnit.Case
+
+  alias StellarBase.XDR.{
+    SCSpecTypeDef,
+    SCSpecType,
+    SCSpecTypeVec,
+    SCSpecTypeMap,
+    SCSpecTypeSet,
+    SCSpecTypeTuple,
+    SCSpecTypeBytesN,
+    SCSpecTypeUDT,
+    SCSpecTypeOption,
+    SCSpecTypeResult,
+    SCSpecTypeDef12,
+    UInt32,
+    Void,
+    String60
+  }
+
+  describe "SCSpecTypeDef" do
+    setup do
+      discriminants = [
+        %{
+          status_type: SCSpecType.new(:SC_SPEC_TYPE_VAL),
+          sc_code: Void.new(),
+          binary: <<0, 0, 0, 0>>
+        },
+        %{
+          status_type: SCSpecType.new(:SC_SPEC_TYPE_U32),
+          sc_code: Void.new(),
+          binary: <<0, 0, 0, 1>>
+        },
+        %{
+          status_type: SCSpecType.new(:SC_SPEC_TYPE_I32),
+          sc_code: Void.new(),
+          binary: <<0, 0, 0, 2>>
+        },
+        %{
+          status_type: SCSpecType.new(:SC_SPEC_TYPE_U64),
+          sc_code: Void.new(),
+          binary: <<0, 0, 0, 3>>
+        },
+        %{
+          status_type: SCSpecType.new(:SC_SPEC_TYPE_I64),
+          sc_code: Void.new(),
+          binary: <<0, 0, 0, 4>>
+        },
+        %{
+          status_type: SCSpecType.new(:SC_SPEC_TYPE_U128),
+          sc_code: Void.new(),
+          binary: <<0, 0, 0, 5>>
+        },
+        %{
+          status_type: SCSpecType.new(:SC_SPEC_TYPE_I128),
+          sc_code: Void.new(),
+          binary: <<0, 0, 0, 6>>
+        },
+        %{
+          status_type: SCSpecType.new(:SC_SPEC_TYPE_BOOL),
+          sc_code: Void.new(),
+          binary: <<0, 0, 0, 7>>
+        },
+        %{
+          status_type: SCSpecType.new(:SC_SPEC_TYPE_SYMBOL),
+          sc_code: Void.new(),
+          binary: <<0, 0, 0, 8>>
+        },
+        %{
+          status_type: SCSpecType.new(:SC_SPEC_TYPE_BITSET),
+          sc_code: Void.new(),
+          binary: <<0, 0, 0, 9>>
+        },
+        %{
+          status_type: SCSpecType.new(:SC_SPEC_TYPE_STATUS),
+          sc_code: Void.new(),
+          binary: <<0, 0, 0, 10>>
+        },
+        %{
+          status_type: SCSpecType.new(:SC_SPEC_TYPE_BYTES),
+          sc_code: Void.new(),
+          binary: <<0, 0, 0, 11>>
+        },
+        %{
+          status_type: SCSpecType.new(:SC_SPEC_TYPE_ADDRESS),
+          sc_code: Void.new(),
+          binary: <<0, 0, 0, 13>>
+        },
+        %{
+          sc_code:
+            SCSpecTypeOption.new(SCSpecTypeDef.new(Void.new(), SCSpecType.new(:SC_SPEC_TYPE_VAL))),
```

Comment:
This is exactly the problem the pipeline seeks to solve. Several cases with the same issue, please fix!

https://elixirschool.com/en/lessons/basics/pipe_operator#introduction-0